scripts/seastar-json2code.py

```diff
@@ -370,7 +370,13 @@ def is_model_valid(name, model):
         return ""
     properties = getitem(model[name], "properties", name)
     for var in properties:
-        type = getitem(properties[var], "type", name + ":" + var)
+        try:
+            type = getitem(properties[var], "type", name + ":" + var)
+        except Exception as e:
+            try:
+                type = getitem(properties[var], "$ref", name + ":" + var)
+            except:
+                raise e
```

Comment:
if `$ref` is the notation supported by swagger spec, probably we should check if first. and i'd suggest just use `dict.get()` . simpler this way, like:

```py
# to be backward compatible, support "type" as well
type = (properties[var].get("$ref") or
        properties[var].get("type"))
if type is None:
  raise KeyError(f"'$ref' not found in {name}:{var}")
```